### PR TITLE
Fix for APIMANAGER-2994

### PR DESCRIPTION
**Pull request by praminda**

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/listeners/KeyManagerUserOperationListener.java [hunk 1]
```diff
@@ -61,32 +71,68 @@ public int getExecutionOrderId() {
...
+            } else {
+                username = IdentityUtil.addDomainToName(username, userDomain);
+            }
+
+            WorkflowExecutor userSignupWFExecutor = WorkflowExecutorFactory.getInstance().
+                    getWorkflowExecutor(WorkflowConstants.WF_TYPE_AM_USER_SIGNUP);
+            String workflowExtRef = apiMgtDAO.getExternalWorkflowReferenceForUserSignup(username);
+            userSignupWFExecutor.cleanUpPendingTask(workflowExtRef);
+        } catch (WorkflowException e) {
+            log.error("Error while cleaning up workflow task for the user: " + username, e);
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/listeners/KeyManagerUserOperationListener.java]: any reason only logging the exception?

File: components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/listeners/KeyManagerUserOperationListener.java [hunk 2]
```diff
@@ -61,32 +71,68 @@ public int getExecutionOrderId() {
...
+        } catch (APIManagementException e) {
+            log.error("Error while cleaning up workflow task for the user: " + username, e);
+        } catch (org.wso2.carbon.user.api.UserStoreException e) {
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/listeners/KeyManagerUserOperationListener.java]: method signature says it's throws userstore exception, it's caught here?

File: components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/listeners/KeyManagerUserOperationListener.java [hunk 3]
```diff
@@ -61,32 +71,68 @@ public int getExecutionOrderId() {
...
-                                                                    throws UserStoreException {
+    public boolean doPreDeleteUser(String username, UserStoreManager userStoreManager) throws UserStoreException {
 
+        boolean isTenantFlowStarted = false;
+        ApiMgtDAO apiMgtDAO = new ApiMgtDAO();
+        try {
+            String tenantDomain = MultitenantUtils.getTenantDomain(APIUtil.replaceEmailDomainBack(username));
+
+            if (tenantDomain != null && !MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain)) {
+                isTenantFlowStarted = true;
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/listeners/KeyManagerUserOperationListener.java]: tenant flow doesn't start yet

File: components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/listeners/KeyManagerUserOperationListener.java (hunk 1 above)
Comment (praminda) [inline components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/listeners/KeyManagerUserOperationListener.java]: Comments added in https://github.com/praminda/carbon-apimgt/commit/06b8f56a8eb0fa81126c41fdce3f41e90a87b5c5

File: components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/listeners/KeyManagerUserOperationListener.java (hunk 2 above)
Comment (praminda) [inline components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/listeners/KeyManagerUserOperationListener.java]: Updated method signature in https://github.com/praminda/carbon-apimgt/commit/06b8f56a8eb0fa81126c41fdce3f41e90a87b5c5

File: components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/listeners/KeyManagerUserOperationListener.java (hunk 3 above)
Comment (praminda) [inline components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/listeners/KeyManagerUserOperationListener.java]: Fixed in https://github.com/praminda/carbon-apimgt/commit/06b8f56a8eb0fa81126c41fdce3f41e90a87b5c5

**Pull request merged**